File: web/backend/app/schemas/inbox.py
from typing import Optional
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel

class InboxThreadResponse(BaseModel):
    id: UUID
    phone: str
    agent_id: UUID
    contact_name: Optional[str] = None
    mode: str  # "fyi" or "action"
    note: Optional[str] = None
    forwarded_by_name: str
    forwarded_by_picture: Optional[str] = None
    has_read: bool = False
    is_archived: bool = False
    last_message_text: Optional[str] = None
    last_message_at: Optional[datetime] = None
    created_at: datetime

class ThreadDetailResponse(BaseModel):
    thread: InboxThreadResponse
    messages: list['MessageResponse']

class ReplyRequest(BaseModel):
    text: str = ""
    attachment_type: Optional[str] = None
    attachment_url: Optional[str] = None

class ReactRequest(BaseModel):
    message_guid: str
    reaction_type: str  # love, like, dislike, laugh, emphasize, question

# Forward import
from app.schemas.message import MessageResponse
ThreadDetailResponse.model_rebuild()

File: web/frontend/src/api/inbox.ts
import { api } from './client';
import type { Message } from './conversations';

export interface InboxThread {
  id: string;
  phone: string;
  agent_id: string;
  contact_name: string | null;
  mode: 'fyi' | 'action';
  note: string | null;
  forwarded_by_name: string;
  forwarded_by_picture: string | null;
  has_read: boolean;
  last_message_text: string | null;
  last_message_at: string | null;
  created_at: string;
}

export interface ThreadDetail {
  thread: InboxThread;
  messages: Message[];
}

export const inboxApi = {
  list: () => api.get<InboxThread[]>('/inbox'),
  get: (threadId: string) => api.get<ThreadDetail>(`/inbox/${threadId}`),
  reply: (threadId: string, text: string, attachmentType?: string, attachmentUrl?: string) =>
    api.post(`/inbox/${threadId}/reply`, { text, attachment_type: attachmentType, attachment_url: attachmentUrl }),
  react: (threadId: string, messageGuid: string, reactionType: string) =>
    api.post<{ status: string }>(`/inbox/${threadId}/react`, { message_guid: messageGuid, reaction_type: reactionType }),
  markRead: (threadId: string) => api.patch(`/inbox/${threadId}/read`, {}),
};
